tokio/src/runtime/scheduler/multi_thread/worker.rs

```diff
@@ -612,7 +616,29 @@ impl Core {
         if self.tick % worker.handle.shared.config.global_queue_interval == 0 {
             worker.inject().pop().or_else(|| self.next_local_task())
         } else {
-            self.next_local_task().or_else(|| worker.inject().pop())
+            let maybe_task = self.next_local_task();
+
+            if maybe_task.is_some() {
+                return maybe_task;
+            }
+
+            let cap = self.run_queue.capacity();
+
+            // The worker is currently idle, pull a batch of work from the
+            // injection queue. We don't want to pull *all* the work so other
+            // workers can also get some.
+            let n = cmp::min(
+                worker.inject().len() / worker.handle.shared.remotes.len() + 1,
```

Comment:
note: we should do experiments here to see what proportions of tasks work well in the wild

Reply:
The constants here were copied from Go.

Reply:
We can do our own experiments, but this is a good starting point.